### ISSUE TITLE
Bad state: Future already completed

### ISSUE THREAD

**Issue by vfiruz97** [schultek/stormberry#69]
Hi **Kilian Schulte**!
First I would like to thank you for this package. It seems to me better package among its concurrents especially when we need relationships. 

here some suggestion from me. may be they are already implemented but I could not find them.  

- [ ] default value in models accessors (to fill updatedAt, createdAt automatically and many other cases)
- [ ] using postgresql data types in models (now string converts text which is unwanted in many cases)
- [ ] Queries & Actions classes with table name in it (e. db.alerts.query(query, params) I would like to have 'alerts' in query without passing it)

Here the issue which makes me create an issue I noticed the strange behaviour. 

```dart
@Model(
  indexes: [
    TableIndex(name: 'user_id_alerts_index', columns: ['user_id'])
  ],
  views: [#Complete, #Reduced],
)
abstract class Alert {
  @PrimaryKey()
  @AutoIncrement()
  int get id;

  @HiddenIn(#Reduced)
  User get user;

  String get message;
  String? get path;
  bool get isRead;
  bool get isImportant;
  int get statusIndex;
  DateTime get createdAt;
  DateTime get updatedAt;
}
``` 

```dart
final db = Database(
  host: '127.0.0.1',
  port: 5432,
  database: 'xyz',
  user: 'xyz',
  password: 'xyz',
  useSSL: false,
);

void main() async {
  try {
    seeds();
  } catch (e) {
    print(e);
  }
}

AlertInsertRequest _getAlertInsertRequestInstance({
  required int userId,
  required String message,
  required int statusIndex,
  required bool isImportant,
  String? path,
}) {
  return AlertInsertRequest(
    userId: userId,
    message: message,
    statusIndex: statusIndex,
    isImportant: isImportant,
    path: path,
    isRead: false,
    updatedAt: DateTime.now().toUtc(),
    createdAt: DateTime.now().toUtc(),
  );
}

void seeds() async {
  firstInsert();
  secondInsert();
}

void firstInsert() async {
  var notId = await db.alerts.insertOne(
    _getAlertInsertRequestInstance(
      userId: 1,
      message: 'Ride #3 rejected.',
      statusIndex: NotificationStatus.warning.v,
      isImportant: true,
    ),
  );
  print('1: $notId');
}

void secondInsert() async {
  var notId = await db.alerts.insertOne(
    _getAlertInsertRequestInstance(
      userId: 2,
      message: 'Passenger #1 rejected from ride #3.',
      statusIndex: NotificationStatus.warning.v,
      isImportant: false,
    ),
  );
  print('2: $notId');
}
``` 

<details><summary>Output Details</summary>
<p>

Bad state: Future already completed
#0      _AsyncCompleter.complete (dart:async/future_impl.dart:41:31)
#1      Database.cancelTransaction (package:stormberry/src/core/database.dart:139:27)
#2      Database.runTransaction (package:stormberry/src/core/database.dart:171:7)
<asynchronous suspension>
#3      secondInsert (file:///Users/vfiruz/Desktop/rohtj/backend/rohtj_server_grpc/seed/test.dart:58:15)
<asynchronous suspension>

</p>
</details> 

if I comment out secondInsert then it works.
If I make the returning type Future for firstInsert, secondInsert functions and use await before them then it works
if you could to check it out or explain what was the reason.

The reason not to make future and await sequently is that I want to reduce server response. 
run someInserts without waiting and return server response.
and wondering to use some instance of Database class, is it possible? 

Thanks!

**Comment by schultek** [comment]
Thanks, this seems like a bug with handling the transactions. I will take a look (prob next week).

**Comment by schultek** [comment]
This is fixed in the new version (`0.14.0`). There the generated methods no longer run their own transaction, instead you manage the transaction yourself and can call multiple operations in one transaction.
